### PR TITLE
feat: add dark mode to jx-docs website

### LINKED ISSUE #2877: Feature request: Add a dark mode to the website
It would be nice to have a dark mode. This would help in viewing the docs for a much longer amount of time (less strain on the eyes and battery)
This can be a good starting point: https://yonkov.github.io/post/add-dark-mode-toggle-to-hugo/

### PR DESCRIPTION
**Pull request by ysahil97**
fixes #2877

### CODE REVIEW

File: layouts/partials/toggle.html [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+<script>
+    var body = document.body;
+    var aref = document.getElementsByClassName('td-sidebar-link__section');
+    var are = document.getElementsByClassName('td-sidebar-link td-sidebar-link__page');
+    var anav = document.getElementsByClassName('nav-link');
+    var intro_bg = document.getElementsByClassName('py-5 bg-light');
+    var paperclip = document.getElementsByClassName('fa-paperclip');
+
+    var elems = [aref,are,anav,intro_bg,paperclip];
+	var switcher = document.getElementsByClassName('js-toggle')[0];
```
Comment (ankitm123) [inline layouts/partials/toggle.html:168]: Would be good to format them for readability purposes :+1:

**Comment by ankitm123** [comment]
@ysahil97 great work :100: 
Would be nice to fix the issue with dark mode not being persisted when navigating from one page to the other.

File: layouts/partials/toggle.html [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+    //     for (i = 0; i < anav.length; i++) {
+    //       anav[i].classList.add('dark-mode');
+    //     }
+    //     for (i = 0;i<intro_bg.length; i++){
+    //       intro_bg[i].classList.add('dark-mode');
+    //     }
+    //     paperclip[0].classList.add('dark-mode');
+	// }
+
+</script>
```
Comment (ankitm123) [inline layouts/partials/toggle.html:231]: Also missing new line.

File: layouts/partials/toggle.html [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+                }
+            }
+
+			setTimeout(function() {
+				localStorage.removeItem('darkMode');
+			}, 100);
+		}
+	})
+
+	//Check Storage. Keep user preference on page reload
```
Comment (ankitm123) [inline layouts/partials/toggle.html:201]: This is important to fix, else it limits how useful this feature can be.

File: layouts/partials/toggle.html (hunk 1 above)
Comment (ysahil97) [reply layouts/partials/toggle.html]: Accepted, these changes are now done

File: layouts/partials/toggle.html (hunk 2 above)
Comment (ysahil97) [reply layouts/partials/toggle.html]: Dark mode scripts are moved to scripts.html partial. Also the end of this file has newline inserted.

**Comment by ankitm123** [comment]
@MarckK any idea why the build is failing?

**Comment by MarckK** [comment]
Great work, @ysahil97! This looks good 👍 

Should the toggle be moved over to the right on the top nav bar, to be with the site navigation links, and not the branding links of Jenkins X and CDFoundation? 
Screenshot 2020-09-16 at 09 29 31

We could put it next to the language ('English') toggle link? It could be on either side, but next to the Search box may look better.

**Comment by ysahil97** [comment]
Sure @MarckK, let's place it in between english toggle link and search box.

**Comment by MarckK** [comment]
This looks good! Thank you, @ysahil97 

/lgtm

**Pull request merged**